src/main/java/com/thealgorithms/searches/SquareRootBinarySearch.java

```diff
@@ -37,24 +41,48 @@ public static void main(String args[]) {
      * @param num Number
      * @return answer
      */
-    private static long squareRoot(long num) {
+    private static double squareRoot(long num, int precision) {
         if (num == 0 || num == 1) {
             return num;
         }
-        long l = 1;
-        long r = num;
-        long ans = 0;
+        long l = 0, r = num;
+
         while (l <= r) {
-            long mid = l + (r - l) / 2;
-            if (mid == num / mid) {
+
+            long mid = l + (r - l)/2;
+
+            if (mid * mid == num) {
                 return mid;
-            } else if (mid < num / mid) {
-                ans = mid;
-                l = mid + 1;
-            } else {
+            }
+
+            else if (mid * mid >= num)
                 r = mid - 1;
+
+            else
+                l = mid + 1;
+        }
+
+        return decimal(num, (double)r, precision);
+    }
+
+    /*
+     * This function takes the calculated floor value and calculates the decimal value
+     * of the root as per the provided precision value.
+     */
+
+    private static double decimal(long n, double root, int precision) {
```

Comment:
Why is it called `decimal`?

Reply:
Because it's giving the decimal value till the provided precision value. Shall I change it to something else?